helpers/test/test_lib_tasks_docker_release.py

```diff
@@ -182,3 +179,250 @@ def test_docker_build_multi_arch(self) -> None:
             remove_lead_trail_empty_lines=True,
             dedent=True,
         )
+
+
+# #############################################################################
+# TestDockerBuildProdImage1
+# #############################################################################
+
+
+class TestDockerBuildProdImage1(hunitest.TestCase):
+
+    def setUp(self) -> None:
+        """
+        Set up test environment and initialize all necessary mocks.
+        """
+        super().setUp()
+        # Mock system calls.
+        self.system_patcher = umock.patch("helpers.hsystem.system")
+        self.mock_system = self.system_patcher.start()
+        # Mock run.
+        self.run_patcher = umock.patch("helpers.lib_tasks_utils.run")
+        self.mock_run = self.run_patcher.start()
+        # Mock version validation.
+        self.version_patcher = umock.patch(
+            "helpers.lib_tasks_docker.dassert_is_subsequent_version"
+        )
+        self.mock_version = self.version_patcher.start()
+        # Mock docker login.
+        self.docker_login_patcher = umock.patch(
+            "helpers.lib_tasks_docker.docker_login"
+        )
+        self.mock_docker_login = self.docker_login_patcher.start()
+        # Mock environment variable.
+        self.env_patcher = umock.patch.dict(
+            "os.environ", {"CSFY_ECR_BASE_PATH": "dummy.ecr.path"}
+        )
+        self.env_patcher.start()
+
+    def tearDown(self) -> None:
+        """
+        Clean up test environment by stopping all mocks after each test case.
+        """
+        self.system_patcher.stop()
+        self.run_patcher.stop()
+        self.version_patcher.stop()
+        self.docker_login_patcher.stop()
+        self.env_patcher.stop()
+        super().tearDown()
+
+    def test_docker_build_prod_image(self) -> None:
+        """
+        Test building a prod Docker image with single architecture.
+
+        This test verifies that the correct sequence of commands is
+        generated for building a prod Docker image.
+        """
+        # Prepare inputs.
+        mock_ctx = httestlib._build_mock_context_returning_ok()
+        test_version = "1.0.0"
+        test_base_image = "test-registry.com/test-image"
+        # Call tested function.
+        hltadore.docker_build_prod_image(
+            mock_ctx,
+            test_version,
+            base_image=test_base_image,
+            cache=False,
+        )
+        actual_cmds = _extract_commands_from_call(self.mock_run.call_args_list)
```

Comment:
You can create a function with the common idiom for checking the output

def check_docker_command_output(self_: Any, exp: str, ...)